src/util.cc

```diff
@@ -423,21 +427,82 @@ const char* SpellcheckString(const char* text, ...) {
 }
 
 #ifdef _WIN32
+std::wstring Utf8ToWide(const std::string& u8_string) {
+  if (u8_string.empty())
+    return std::wstring();
+
+  size_t size_needed =
+      MultiByteToWideChar(CP_UTF8, 0, u8_string.data(),
+                          static_cast<int>(u8_string.size()), NULL, 0);
+  if (size_needed == 0)
+    Fatal("Failed conversion to wide string");
+
+  vector<wchar_t> w_buffer(size_needed);
+  int w_size_needed = MultiByteToWideChar(CP_UTF8, 0, u8_string.data(),
+                                          static_cast<int>(u8_string.size()),
+                                          &w_buffer[0], w_buffer.size());
```

Comment:
w_buffer.data()

Reply:
As strings are null terminated, this shouldn't make a difference.

Reply:
This is not true.

C++03 strings are not guaranteed to be contiguous, and may be reference counted / copy-on-write.

std::string::data() provides a contiguous buffer to read from.

Also, the size being provided for both string parameters means that the API doesn't check for a nul terminator. It reads/write up to size. https://docs.microsoft.com/en-us/windows/win32/api/stringapiset/nf-stringapiset-multibytetowidechar

Reply:
However, as discussed in another comment, I was mistaken about the usability of the data() function.

&w_buffer[0] is the only option (requires that we assume C++11 for Windows builds)

Please disregard my suggestion here.

Reply:
> C++03 strings are not guaranteed to be contiguous, and may be reference counted / copy-on-write.

This code is compiled in C++11 mode and so is std::string::data().